packages/medusa/src/api/routes/store/carts/create-cart.ts

```diff
@@ -186,19 +200,53 @@ export default async (req, res) => {
       const createdCart = await cartServiceTx.create(toCreate)
 
       if (validated.items?.length) {
-        const generateInputData = validated.items.map((item) => {
-          return {
-            variantId: item.variant_id,
-            quantity: item.quantity,
-          }
-        })
-        const generatedLineItems: LineItem[] = await lineItemServiceTx.generate(
-          generateInputData,
-          {
-            region_id: regionId,
-            customer_id: req.user?.customer_id,
-          }
-        )
+        let generatedLineItems: LineItem[]
+
+        if (
+          featureFlagRouter.isFeatureEnabled(
+            IsolateProductDomainFeatureFlag.key
+          )
+        ) {
+          const variantIds = validated.items
+            .map((i) => i.variant_id)
+            .filter(Boolean)
+
+          const remoteQuery = req.scope.resolve("remoteQuery")
+
+          const variants = await retrieveVariantsWithIsolatedProductModule(
+            remoteQuery,
+            variantIds
+          )
+
+          const generateInputData = validated.items.map((item) => {
+            const variant = variants.find((v) => v.id === item.variant_id)!
+
+            return prepareLineItemData(variant, item.quantity)
+          })
+
+          generatedLineItems = await lineItemServiceTx.generateLineItem(
+            generateInputData,
```

Comment:
question: What happens if no variants with provided IDs exist and are passed here?

Reply:
We get an empty array. I created a validation util for this that is used in the workflow [here](https://github.com/medusajs/medusa/pull/5148/files#diff-b674642bc6f1c34458a9e86095a82d1424f825ec84f6f057fc2d6f4879e52e44R47). We should add this here as well, wdyt?